packages/lib/jobs/client/_internal/job.ts

```diff
@@ -0,0 +1,58 @@
+import { z } from 'zod';
+
+import type { Json } from './json';
+
+export type SimpleTriggerJobOptions = {
+  id?: string;
+  name: string;
+  payload: unknown;
+  timestamp?: number;
+};
+
+export const ZSimpleTriggerJobOptionsSchema = z.object({
+  id: z.string().optional(),
+  name: z.string(),
+  payload: z.unknown().refine((x) => x !== undefined, { message: 'payload is required' }),
+  timestamp: z.number().optional(),
+});
```

Comment:
Got a small gripe with this since the types actually mismatch

Can we union a bunch of types to create the unknown without undefined so we can infer this schema type for `SimpleTriggerJobOptions`?

Then this can be updated

```
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        const options = await json(req)
          .then(async (data) => ZSimpleTriggerJobOptionsSchema.parseAsync(data))
          // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
          .then((data) => data as SimpleTriggerJobOptions)
          .catch(() => null);
```

Reply:
The simple type is used as a placeholder within the generic provider where the covering type of `TriggerJobOptions` provides the actual type safety for usage of the client.